src/CoinbaseSmartWalletFactory.sol

```diff
@@ -31,24 +31,22 @@ contract CoinbaseSmartWalletFactory {
             revert OwnerRequired();
         }
 
-        (bool alreadyDeployed, address accountAddress) =
-            LibClone.createDeterministicERC1967(msg.value, implementation, _getSalt(owners, nonce));
+        bytes32 salt = _getSalt(owners, nonce);
+        address accountAddress = Clones.predictDeterministicAddress(implementation, salt, address(this));
 
         account = CoinbaseSmartWallet(payable(accountAddress));
 
-        if (alreadyDeployed == false) {
-            account.initialize(owners);
+        if (address(account).code.length > 0) {
+            return account;
         }
+
+        Clones.cloneDeterministic(implementation, salt);
+        account.initialize{value: msg.value}(owners);
```

Comment:
msg.value kinda has unpredictable behavior here, in that if the account is already deployed, this will just sit on the factory. Maybe should just remove payable on this function

Reply:
What about returning the eth if already existing?